tests/functional/local.env

```diff
@@ -9,4 +9,5 @@ export ANCHORE_TEST_DOCKER_REGISTRY_HOST=localhost:5000
 export ANCHORE_BASE_URL=http://localhost:8228/v1
 export ANCHORE_CLI_URL=http://localhost:8228/v1
 export ANCHORE_CLI_USER=admin
-export ANCHORE_CLI_PASS=foobar
\ No newline at end of file
+export ANCHORE_CLI_PASS=foobar
+export TOX_ENV=py36
```

Comment:
this abstraction is making it obscure how/where/what version of Python we are testing. The levels of indirection look like:

1. Makefile
2. scripts/ci/test-functional
3. tox
4. local.env

Someone running functional tests now needs to be aware of a spot that sets these, which is very surprising to anyone not used to running these tests with make

Reply:
It would be nice to get some time to simplify the entire setup, but, this PR is not introducing any of these abstractions (introduced https://github.com/anchore/anchore-engine/pull/679 ). 

Background: The reason for the change is due to the fact that the functional tests are not run within the build docker image, so dependencies were being independently pinned and installed in different places (skopeo, anchore-cli, etc), leading to a fundamental issue when introducing syft. The location where the tests were running needed to be based on the same production artifact being produced (not running on a CI host).

As an added nicety for folks that wanted to run functional tests locally the `local.env` was added --not used in CI. In CI the job parameters control which python version is used (introduced https://github.com/anchore/anchore-engine/pull/698 , and was a conflict in the rebase). 

There was team consensus on this approach and it's use is documented within `DEVELOPING.md` (which I've updated here as well).

Reply:
> It would be nice to get some time to simplify the entire setup, but, this PR is not introducing any of these abstractions (introduced #679 ).

I understand these got introduced previously because I raised my concerns in that PR as well. What my suggestion is here is to _avoid adding more_ to this file in favor of defaults. In this particular case, the default are already one abstraction level up, in the script that gets called by `make`.

> 
> Background: The reason for the change is due to the fact that the functional tests are not run within the build docker image, so dependencies were being independently pinned and installed in different places (skopeo, anchore-cli, etc), leading to a fundamental issue when introducing syft. The location where the tests were running needed to be based on the same production artifact being produced (not running on a CI host).

I agree that having dependencies installed in a normalized and repeatable way within the container that is getting tested is the way to go.

> 
> As an added nicety for folks that wanted to run functional tests locally the `local.env` was added --not used in CI. In CI the job parameters control which python version is used (introduced #698 , and was a conflict in the rebase).
> 

Right, this looks like a developer-only change, which is not related to getting tests passing on CI. As a tox user, I do know how to run tests and set the particular Python version I need. Having to source a file that sets that for me is surprising to say the least - even if documented.

Reply:
I hear you regarding sane defaults as a nicety --right now this branch took out multiple layers of defaults that were hiding and caused hours of troubleshooting, if we put them back in my only request is to try to put the defaults in a single spot (wherever that should be).

Please feel free to change this in a way that makes more sense! I'm all for it.